docs/assertions.md

```diff
@@ -177,6 +186,20 @@ TEST_CASE_METHOD((Fixture<int, int>), "foo", "[bar]") {
 This solution is not always applicable, because it might require extra
 changes on the Catch's side to work.
 
+## Lambdas
+Lambdas are a C++ language feature and can be freely used inside a `TEST_CASE` or a `SECTION`.
+
+```cpp
+#include <catch2/catch_test_macros.hpp>
+
+TEST_CASE("Dumb") {
+    auto fnTest = [](const bool bPass) { CHECK(bPass); };
+
+    fnTest(true);
+}
+```
+[godbolt](https://catch2.godbolt.org/z/ebdr9vKcj)
+
```

Comment:
This shows, but doesn't explain to non-experts, an important feature: the fact that assertions can happen outside the `TEST_CASE`.

There is also something interesting to be said about lambdas specifically: how do their capture lists behave across multiple invocations (due to `SECTION`s).

So I would like to see a bit of text first explaining assertions in separate functions, then another piece about lambdas.